### PR TITLE
Fetch and display featured images

### LINKED ISSUE #8: Display the featured image on each post
This isn't implemented yet.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #8 

---

This patch is mainly about getting the data and displaying the featured image. Also, this is going to be interesting. I don't know how to do that properly without a CSS background...

## Screenshots

WP theme:

Screenshot_2021-03-30 Example of an add-on card - Mozilla Add-ons Blog

Blog:

Screenshot_2021-03-30 Example of an add-on card - Mozilla Add-ons Blog(1)

### CODE REVIEW

File: eleventy.config.js [hunk 1]
```diff
@@ -56,6 +56,15 @@ module.exports = function configure(eleventyConfig) {
     });
   });
 
+  eleventyConfig.addFilter('mediaGetFullURL', (allMedia, featuredMediaId) => {
+    const media = allMedia.find((item) => {
+      return featuredMediaId === item.id;
+    });
+
+    // What could possibly go wrong? :/
+    return media ? media.media_details.sizes.full.source_url : '';
```
Comment (muffinresearch) [inline eleventy.config.js:65]: This is a bit YDD (YOLO Driven Development). Could you be more defensive or is this data structure guaranteed?

File: src/layouts/base.njk [hunk 2]
```diff
@@ -16,7 +16,15 @@
     {% endblock %}
   </head>
   <body>
-    {% include "header.njk" %}
+    <header
+      class="Header"
+      style="background: #20123a url('{{ featuredImage }}') no-repeat center;">
```
Comment (muffinresearch) [inline src/layouts/base.njk:21]: This will require unsafe-inline for the CSP. Is there no way to do this with an inline image?

**Review by muffinresearch — APPROVED** [review]
r+wc

File: eleventy.config.js (hunk 1 above)
Comment (willdurand) [reply eleventy.config.js:65]: > This is a bit YDD (YOLO Driven Development). Could you be more defensive or is this data structure guaranteed?

This is the structure we get from the API. I'll add tests and some checks.

File: src/layouts/base.njk (hunk 2 above)
Comment (willdurand) [reply src/layouts/base.njk:21]: I don't know. I guess we'll see once we implement the final design.

File: eleventy.config.js (hunk 1 above)
Comment (willdurand) [reply eleventy.config.js]: I added a helper function to get the size of a "media" and it's covered by tests.

**Pull request merged**